sdk/compiler/damlc/daml-compiler/src/DA/Daml/Compiler/DataDependencies.hs

```diff
@@ -330,7 +332,7 @@ generateSrcFromLf env = noLoc mod
 
     genExports :: Gen [LIE GhcPs]
```

Comment:
You could maybe add a toplevel comment on that function explaining the overall strategy:
- re-exports are read from $$export via classReexports
- explicit exports are read from $$explicitExport (and $$explicitExports) unless usesExplicitExports is False, in which case we export everything via selfReexport
- exports/reexports are ignored in some special cases: internal names, ghc-prim, class method names